index.js

```diff
@@ -55,71 +56,49 @@ function GCSAdapter() {
   this._bucketPrefix = options.bucketPrefix;
   this._directAccess = options.directAccess;
 
-  this._gcsClient = new storage(options);
+  this._gcsClient = new Storage(options);
 }
 
 GCSAdapter.prototype.createFile = function(filename, data, contentType) {
   let params = {
     metadata: {
-      contentType: contentType || 'application/octet-stream'
+	contentType: contentType || 'application/octet-stream'
     }    
   };
 
-  return new Promise((resolve, reject) => {
-    let file = this._gcsClient.bucket(this._bucket).file(this._bucketPrefix + filename);
-    // gcloud supports upload(file) not upload(bytes), so we need to stream.
-    var uploadStream = file.createWriteStream(params);
-    uploadStream.on('error', (err) => {
-      return reject(err);
-    }).on('finish', () => {
-      // Second call to set public read ACL after object is uploaded.
-      if (this._directAccess) {
-        file.makePublic((err, res) => {
-          if (err !== null) {
-            return reject(err);
-          }
-          resolve();
-        });
-      } else {
-        resolve();
-      }
+  async function uploadFile() {
+    await this._gcsClient.bucket(this._bucket).upload(this._bucketPrefix + filename, {
+	gzip: true,
+	params,
     });
-    uploadStream.write(data);
-    uploadStream.end();
-  });
+
+    console.log(`${filename} uploaded to ${bucketName}.`);
+  }
+
+  uploadFile().catch(console.error);
```

Comment:
I believe we should await. And also not only output the error to the console but let it be thrown.